Composer/packages/server/src/models/asset/assetManager.ts

```diff
@@ -92,58 +96,78 @@ export class AssetManager {
     return ref;
   }
 
-  private async getRemoteTemplate(template: BotTemplate, destinationPath: string) {
-    // install package
-    if (template.package) {
-      const { stderr: initErr } = await execAsync(`dotnet new -i ${template.package.packageName}`);
-      if (initErr) {
-        throw new Error(initErr);
+  public async copyRemoteProjectTemplateToV2(
+    templateId: string,
+    projectName: string,
+    ref: LocationRef,
+    user?: UserIdentity,
+    locale?: string
+  ): Promise<LocationRef> {
+    try {
+      // user storage maybe diff from template storage
+      const dstStorage = StorageService.getStorageClient(ref.storageId, user);
+      const dstDir = Path.resolve(ref.path);
+      if (await dstStorage.exists(dstDir)) {
+        log('Failed copying template to %s', dstDir);
+        throw new Error('already have this folder, please give another name');
       }
-      const { stderr: initErr2 } = await execAsync(`dotnet new ${template.id}`, {
-        cwd: destinationPath,
-      });
-      if (initErr2) {
-        throw new Error(initErr2);
+
+      mkdirSync(dstDir, { recursive: true });
+
+      // find selected template
+      const npmPackageName = templateId;
+      const generatorName = npmPackageName.toLowerCase().replace('generator-', '');
+
+      const remoteTemplateAvailable = await this.installRemoteTemplate(generatorName, npmPackageName, dstDir);
+
+      if (remoteTemplateAvailable) {
+        await this.instantiateRemoteTemplate(generatorName, dstDir, projectName);
       }
+
+      ref.path = ref.path + `/${projectName}`;
+
+      return ref;
+    } catch {
+      throw new Error('error hit when copying remote template');
+    }
+  }
+
+  private async installRemoteTemplate(generatorName: string, npmPackageName: string, dstDir: string): Promise<boolean> {
+    const registeredGenerators: string[] = await yeomanEnv.getGeneratorNames();
+
+    if (registeredGenerators.indexOf(generatorName) !== -1) {
+      return true;
     } else {
-      throw new Error('selected template has no local or external address');
+      await yeomanEnv.installLocalGenerators({ [npmPackageName]: '*' });
```

Comment:
I don't see how this is ensuring that the locally installed generator is going into our composer data folder.

Perhaps setting the yeoman.cwd will do it?

Reply:
I have now initialized yeomen env with the cwd set to the appData directory, it seems to be working in both web and electron context. I want to verify and see the dir itself in an electron setting. Where would that live when you run yarn start:electron:dev?

Reply:
in your `~/BotFrameworkComposer` folder